Source/JavaScriptCore/runtime/ISO8601.cpp

```diff
@@ -843,7 +909,38 @@ static std::optional<CalendarRecord> parseCalendar(StringParsingBuffer<Character
     if (*buffer != ']')
         return std::nullopt;
     buffer.advance();
-    return CalendarRecord { WTFMove(result) };
+
+    return CalendarRecord { isCritical, false, WTFMove(result) };
+}
+
+template<typename CharacterType>
+static std::optional<Vector<CalendarRecord>>
+parseCalendar(StringParsingBuffer<CharacterType>& buffer)
+{
+// https://tc39.es/proposal-temporal/#prod-Annotations
+//  Annotations :::
+//      Annotation Annotations[opt]
+
+    Vector<CalendarRecord> result;
+    // https://tc39.es/proposal-temporal/#sec-temporal-parseisodatetime
+    bool calendarWasCritical = false;
+    while (canBeCalendar(buffer)) {
+        auto calendarRecord = parseOneCalendar(buffer);
+        if (!calendarRecord)
+            return std::nullopt;
+        result.append(calendarRecord.value());
+        if (calendarRecord->m_critical && result.size() == 1)
+            calendarWasCritical = true;
+        // Check for multiple calendars and critical flag
+        // step 4(a)(ii)(2)(c)(ii)
+        else if (calendarRecord->m_critical || calendarWasCritical)
+            return std::nullopt;
+        // Check for unknown annotations with critical flag
+        // step 4(a)(ii)(2)(d)(i)
+        if (calendarRecord->m_unknown && calendarRecord->m_critical)
+            return std::nullopt;
```

Comment:
Ditto.

Reply:
Likewise, I'm pretty sure this is correct. If there is an unrecognized annotation and it is marked critical (e.g. `[!foo=bar]`), it must be rejected.